static/js/playground-search.js

```diff
@@ -0,0 +1,115 @@
+// Fetch the index.json file from the static folder and create a lunr index.
+let pagesIndex, searchIndex;
+async function initSearchIndex() {
+  try {
+    const res = await fetch("/index.json");
+    pagesIndex = await res.json();
+    searchIndex = lunr(function () {
+      this.field("content");
+      this.field("tags");
+      this.ref("uri");
+      pagesIndex.forEach((page) => this.add(page));
+      console.log("pagesIndex", pagesIndex);
+    });
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+initSearchIndex();
+
+// Add an event listener to the search form to get the search after the user clicks the search button.
+const searchForm = document.getElementById('playground-search-form');
+searchForm.addEventListener('input', (e) => {
+  handleSearchQuery(e);
+});
+
+// Prevent the form from submitting when the user presses the enter key or clicks the submit button.
```

Comment:
The Figma design includes a search icon here, but the search results are displayed as the user types, so this was added for now to prevent the page from reloading. If the icon is removed or if we only display results on click then this can be updated.

Reply:
I added `is_search_enabled` and `is_filter_enabled` to the `/data-playground/_index.md` file and separated both components into separate files to make it easier to toggle between displaying only one, both, or nothing at all.